cmd/crank/beta/migrate/deploymentruntime/converter.go

```diff
@@ -0,0 +1,265 @@
+/*
+Copyright 2024 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package deploymentruntime
+
+import (
+	"errors"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"github.com/crossplane/crossplane/apis/pkg/v1alpha1"
+	"github.com/crossplane/crossplane/apis/pkg/v1beta1"
+)
+
+const (
+	// default container name that XP uses
+	runtimeContainerName = "package-runtime"
+
+	errNilControllerConfig = "ControllerConfig is nil"
+)
+
+var timeNow = time.Now()
+
+// controllerConfigToDeploymentRuntimeConfig converts a ControllerConfig to
+// a DeploymentRuntimeConfig
+func controllerConfigToDeploymentRuntimeConfig(cc *v1alpha1.ControllerConfig) (*v1beta1.DeploymentRuntimeConfig, error) {
+	if cc == nil {
+		return nil, errors.New(errNilControllerConfig)
+	}
+	dt := deploymentTemplateFromControllerConfig(cc)
+	drc := newDeploymentRuntimeConfig(
+		withName(cc.Name),
+		withCreationTimestamp(metav1.NewTime(timeNow)),
```

Comment:
I was surprised as well. It seems its due to https://github.com/kubernetes/kubernetes/issues/109427 which from what I understand was fixed in 1.29? So when we update, we can remove this and other places where its set. Otherwise, we will get `creationTimestamp: null`